src/ui/components/search/autocompletecomponent.js

```diff
@@ -181,12 +186,18 @@ export default class AutoCompleteComponent extends Component {
 
     if (wasOpen && !this._isOpen) {
       this._onClose();
+      this._beforeSearch = true;
     } else if (!wasOpen && this._isOpen) {
       this._onOpen();
     }
 
+    if (queryInputEl.value && this.isQueryInputFocused()) {
+      this._beforeSearch = false;
+    }
+
     super.setState(Object.assign({}, data, {
       hasResults: this.hasResults(data),
+      beforeSearch: this._beforeSearch,
```

Comment:
instead of using this new variable `beforeSearch`, can we use `_isOpen` instead?

Reply:
I tried using that initially, but the state is updated more than once when focus is put on the search bar. so, the first time _isOpen is true, but after that, _isOpen is false even if the user hasn't typed anything yet

Reply:
if we use a `onOpen` variable under the condition `(!wasOpen && this._isOpen)`, that should work. In regard to the toggle value from true to false (update more than once). I see that this is triggered by `DOM.on(queryInput, 'focus',...)` where reset() update state once, and the following autocomplete call to core update state again. Let's change it to use the reset code but without the updateState call

Reply:
updated to use `onOpen` instead